datafusion/expr/src/expr.rs

```diff
@@ -954,11 +957,11 @@ impl Expr {
     }
 
     /// Remove an alias from an expression if one exists.
-    pub fn unalias(self) -> Expr {
-        match self {
-            Expr::Alias(alias) => alias.expr.as_ref().clone(),
-            _ => self,
+    pub fn unalias(&mut self) -> &mut Self {
```

Comment:
Nice spot -- I think we can avoid a copy like this too: https://github.com/apache/arrow-datafusion/pull/8588